org.osgi.test.cases.jdbc/src/org/osgi/test/cases/jdbc/junit/JDBCTestCase.java

```diff
@@ -57,127 +74,155 @@ private static class DfltProps extends Properties {
+				.isInstanceOfAny(String.class, String[].class);
 
+		//then check optional ones
 		Object driverVersion = sr.getProperty(OSGI_JDBC_DRIVER_VERSION);
-
 		if (driverVersion != null) {
 			assertTrue(driverVersion instanceof String,
-					"The DataSourceFactory driver version is not a String");
+					String.format("According to '125.2 Database Driver' the property '%s' must be a String if specified", OSGI_JDBC_DRIVER_VERSION));
 		}
+		Object driverName = sr.getProperty(OSGI_JDBC_DRIVER_NAME);
+		if (driverName != null) {
+			assertTrue(driverName instanceof String,
+					String.format("According to '125.2 Database Driver' the property '%s' must be a String if specified", OSGI_JDBC_DRIVER_NAME));
+		}
+		
 	}
 
+	/**
+	 * This test that 
+	 * <ul>
+	 * <li>all declared capabilities behave according to their contract</li>
+	 * <li>at least one capability is provided</li>
+	 * </ul>
+	 * @param sr the service reference of the {@link DataSourceFactory}
+	 */
 	@ParameterizedTest
-	@ServiceSource(serviceType = DataSourceFactory.class, filter = FILTER_CAPABILITY_CONNECTIONPOOLDATASOURCE)
-	public void testCapabilityConnectionPoolDataSource(DataSourceFactory dsf)
-			throws SQLException {
-
-		assertThatNoException()
-				.isThrownBy(() -> dsf.createConnectionPoolDataSource(null));
-
-		assertThat(dsf.createConnectionPoolDataSource(null)).isNotNull();
-
-		assertThatNoException().isThrownBy(
-				() -> dsf.createConnectionPoolDataSource(new DfltProps()));
-
-		assertThat(dsf.createConnectionPoolDataSource(new DfltProps()))
-				.isNotNull();
+	@ServiceSource(serviceType = DataSourceFactory.class)
+	public void testDeclaredCapabilities(ServiceReference<DataSourceFactory> sr)
+			throws SQLException, InvalidSyntaxException {
+		boolean providesAny = false;
+		for (JdbcCapability<?> capability : ALL_CAPABILITIES) {
+			providesAny |= testCapability(sr, capability);
+		}
+		assertTrue(providesAny,
+				String.format("According to '125.2 Database Driver' the DataSourceFactory must at laest provide one of the follwoing capabilities '%s'.", ALL_CAPABILITIES));
 	}
 
-	@ParameterizedTest
-	@ServiceSource(serviceType = DataSourceFactory.class, filter = FILTER_CAPABILITY_DATASOURCE)
-	public void testCapabilityDataSource(DataSourceFactory dsf)
-			throws SQLException {
-
-		assertThatNoException().isThrownBy(() -> dsf.createDataSource(null));
-
-		assertThat(dsf.createDataSource(null)).isNotNull();
-
-		assertThatNoException()
-				.isThrownBy(() -> dsf.createDataSource(new DfltProps()));
-
-		assertThat(dsf.createDataSource(new DfltProps())).isNotNull();
+	private <T> boolean testCapability(ServiceReference<DataSourceFactory> reference, JdbcCapability<T> capability) throws SQLException, InvalidSyntaxException {
+		DataSourceFactory dsf = bundleContext.getService(reference);
+		try {
+			if (FrameworkUtil.createFilter(String.format("(%s=%s)", OSGI_JDBC_CAPABILITY, capability.capability)).match(reference)) {
```

Comment:
You have a bundle context here. Use it to create the Filter instead of FrameworkUtil.